.github/workflows/build-sample-apps.yml

```diff
@@ -0,0 +1,195 @@
+
+      - name: Generate New Version
+        uses: maierj/fastlane-action@v3.1.0
+        with:
+          subdirectory: apps/${{ matrix.sample-app }}
+          lane: "generate_new_version"
+          options: '{"branch_name":"${{ github.ref_name }}", "pull_request_number":"${{ github.event.pull_request.number }}"}'
+
+      - name: Update Flutter SDK Version
+        uses: maierj/fastlane-action@v3.1.0
+        with:
+          subdirectory: apps/${{ matrix.sample-app }}
+          lane: "update_flutter_sdk_version"
+        env:
+          SDK_VERSION_NAME: ${{ env.SDK_VERSION_NAME }}
+          APP_VERSION_NAME: ${{ env.APP_VERSION_NAME }}
+          APP_VERSION_CODE: ${{ env.APP_VERSION_CODE }}
+
+      - name: Update Sample App Version
+        uses: maierj/fastlane-action@v3.1.0
+        with:
+          subdirectory: apps/${{ matrix.sample-app }}
+          lane: "update_flutter_app_version"
+        env:
+          SDK_VERSION_NAME: ${{ env.SDK_VERSION_NAME }}
+          APP_VERSION_NAME: ${{ env.APP_VERSION_NAME }}
+          APP_VERSION_CODE: ${{ env.APP_VERSION_CODE }}
+
+      - name: Setup workspace credentials in flutter environment files
+        run: |
+          cp ".env.example" ".env"
+          sd 'SITE_ID=.*' "SITE_ID=${{ secrets[format('CUSTOMERIO_{0}_WORKSPACE_SITE_ID', matrix.sample-app)] }}" ".env"
+          sd 'API_KEY=.*' "API_KEY=${{ secrets[format('CUSTOMERIO_{0}_WORKSPACE_API_KEY', matrix.sample-app)] }}" ".env"
+
+      - name: Setup workspace credentials in iOS environment files
+        run: |
+          cp "ios/Env.swift.example" "ios/Env.swift"
+          sd 'siteId: String = ".*"' "siteId: String = \"${{ secrets[format('CUSTOMERIO_{0}_WORKSPACE_SITE_ID', matrix.sample-app)] }}\"" "ios/Env.swift"
+          sd 'apiKey: String = ".*"' "apiKey: String = \"${{ secrets[format('CUSTOMERIO_{0}_WORKSPACE_API_KEY', matrix.sample-app)] }}\"" "ios/Env.swift"
+
+      # Make sure to fetch dependencies only after updating the version numbers and workspace credentials
+
+      - name: Setup flutter environment and install dependencies
+        uses: ./.github/actions/setup-flutter
+
+      - name: Install flutter dependencies for sample app
+        run: flutter pub get
+
+      - name: Setup Android environment for sample app
+        uses: ./.github/actions/setup-android
+
+      - name: Build and upload Android app via Fastlane
+        uses: maierj/fastlane-action@v3.1.0
+        with:
+          subdirectory: apps/${{ matrix.sample-app }}
+          lane: 'android build'
+        env:
+          FIREBASE_APP_DISTRIBUTION_SERVICE_ACCOUNT_CREDS_B64: ${{ secrets.FIREBASE_APP_DISTRIBUTION_SERVICE_ACCOUNT_CREDS_B64 }}
+
+      - name: Setup iOS environment for sample app
+        uses: ./.github/actions/setup-ios
+        with:
+          xcode-version: ${{ env.XCODE_VERSION }}
+
+      - name: Cache CocoaPods downloaded dependencies for faster builds in the future
```

Comment:
instead of compiling android and ios apps inside of the same job, I suggest separating them into their own `job` so they can run in parallel. Could speedup the CI running.

Reply:
This is a good idea as this will not only speed up the process but also prevent the other app not to fail if one of them is not compiling. But the changes might be larger, I'll do them in another PR later after trying it out for React Native apps first.